lib/charms/blackbox_k8s/v0/blackbox_probes.py

```diff
@@ -0,0 +1,652 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Blackbox Probes Library.
+
+## Overview
+
+This document explains how to integrate with the Blackbox charm
+for the purpose of providing a probes metrics endpoint to Prometheus.
+
+## Provider Library Usage
+
+The Blackbox Exporter charm interacts with its datasources using this charm
+library. The goal of this library is to be as simple to use as possible.
+Charms seeking to expose metric endpoints to be probed via Blackbox, must do so
+using the `BlackboxProbesProvider` object from this charm library.
+For the simplest use cases, the BlackboxProbesProvider object requires
+to be instantiated with a list of jobs with the endpoints to monitor.
+A probe in blackbox is defined by a module and a static_config target. Those
+are then organised in a prometheus job for proper scraping.
+The `BlackboxProbesProvider` constructor requires
+the name of the relation over which a probe target
+is exposed to the Blakcbox Exporter charm. This relation must use the
+`blackbox_probes` interface.
+The default name for the metrics endpoint relation is
+`blackbox-targets`. It is strongly recommended to use the same
+relation name for consistency across charms and doing so obviates the
+need for an additional constructor argument. The
+`BlackboxProbesProvider` object may be instantiated as follows
+
+    from charms.blackbox_k8s.v0.blackbox_probes import BlackboxProbesProvider
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        ...
+        self.probes_provider = BlackboxProbesProvider(self, probes=probes_endpoints_config)
```

Comment:
Is there a simple config so short that we could inline it in this example, to better illustrate?